src/expressions/ExecutionSpecificStaticContext.ts

```diff
@@ -60,24 +59,30 @@ export default class ExecutionSpecificStaticContext implements IContext {
 		this._referredVariableByName = Object.create(null);
 		this._referredNamespaceByName = Object.create(null);
 
-		this._variableValueByName = variableByName;
-
 		/**
 		 * This flag will be set to true if this EvaluationContext was used while statically
 		 * compiling a Expression
 		 */
 		this.executionContextWasRequired = false;
 	}
 
-	public getReferredNamespaces(): { namespaceURI: string; prefix: string }[] {
+	public getReferredNamespaces(): { namespaceURI: string; prefix: string }[] | null {
 		return Object.values(this._referredNamespaceByName);
 	}
 
-	public getReferredVariables(): { name: string }[] {
-		return Object.values(this._referredVariableByName);
+	public getReferredVariables(): { name: string }[] | null {
+		const values = Object.values(this._referredVariableByName);
+		if (values.length === 0) {
+			return null;
```

Comment:
Why do we return null instead of the empty array? The array is already allocated so I don't see why we should not use it.

Reply:
mainly to prevent the `.every` on an empty array later on. If the expression does not refer to variables, determining a cache hit is only a single null compare in this case